### PR TITLE
Generalise sidebar campaign published article timeframe

### PR DESCRIPTION
**Pull request by leewynne**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR adds a configurable timeframe for showing articles in the campaign sidebar, for smaller communities, the rate of submission will be much lower therefore this allows a Forem admin to adjust the timeframe so more stories show up in the count.

## Related Tickets & Documents

This is related to issue #11968

## QA Instructions, Screenshots, Recordings

1. Enabled and create a sidebar campaign in admin/config
2. Set the required feature tag for articles to show in the count
3. Enabled the campaign
4. Set the desired campaign article expiry time
5. Set the published time for articles within the required tag to before and after the expiry time
6. Note the change in the Stories count.

### UI accessibility concerns?

No

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

expire

### CODE REVIEW

**Comment by fdoxyz** [comment]
Hey @leewynne thanks for the PR!

The failing specs in Travis are unrelated to the changes here and should've been fixed by another PR merged yesterday, could you try to sync with master and push again please?

File: app/controllers/sidebars_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def show
 
   def get_latest_campaign_articles
     campaign_articles_scope = Article.tagged_with(Campaign.current.featured_tags, any: true)
-      .where("published_at > ? AND score > ?", 4.weeks.ago, 0)
+      .where("published_at > ? AND score > ?", SiteConfig.campaign_articles_expiry_time.weeks.ago, 0)
```
Comment (citizen428) [inline app/controllers/sidebars_controller.rb:14]: **suggestion (non-blocking):** A part of me wants to be really explicit here and call the config option `campaign_article_expiry_in_weeks`.

**Comment by leewynne** [comment]
> Hey @leewynne thanks for the PR!
> 
> The failing specs in Travis are unrelated to the changes here and should've been fixed by another PR merged yesterday, could you try to sync with master and push again please?

Hey Fernando, I think I have done this correctly. How do I restart the travis build?

**Comment by rhymes** [comment]
@leewynne just restarted it, I think only team members can https://travis-ci.com/github/forem/forem/jobs/471754745

**Comment by leewynne** [comment]
> @leewynne just restarted it, I think only team members can https://travis-ci.com/github/forem/forem/jobs/471754745

looks like it failed again, am sure I synced it though :\

**Comment by fdoxyz** [comment]
Sorry, I originally thought this was a problem that was addressed in a PR a while ago but I was mistaken. It seems to be a problem related to some more recent specs that is under investigation (not related to your PR @leewynne).

cc @Ridhwana I think this is another case where the problem with `RELEASE_FOOTPRINT` is coming up. Maybe because it's a PR from a fork and not a branch?

**Comment by Ridhwana** [comment]
@fdoxyz and @leewynne. I'm pushing up a change within the hour to fix this issue, we'll try pulling it into this branch thereafter. I'll ping you once the fix is merged to master.

**Comment by leewynne** [comment]
> @fdoxyz and @leewynne I'm pushing up a change within the hour that I think will fix this flakey specs, we'll try pulling it into this branch thereafter. I'll ping you once the fix is merged to master.

Great thanks!

**Comment by Ridhwana** [comment]
https://github.com/forem/forem/pull/12294 hoping this may solve. Let me know if you're still having this issue once this change has been pulled in :)

**Comment by leewynne** [comment]
> #12294 hoping this may solve. Let me know if you're still having this issue once this change has been pulled in :)



> #12294 hoping this may solve. Let me know if you're still having this issue once this change has been pulled in :)

It failed again :(

**Comment by rhymes** [comment]
@leewynne sorry for the troubles, this time is for a different reason, here's [the PR with the potential fix](https://github.com/forem/forem/pull/12313), we have to evaluate if that's the best strategy moving forward but we're aware of the issue

**Comment by leewynne** [comment]
> @leewynne sorry for the troubles, this time is for a different reason, here's [the PR with the potential fix](https://github.com/forem/forem/pull/12313), we have to evaluate if that's the best strategy moving forward but we're aware of the issue

Ah great @rhymes - as soon as it get's merge, I'll sync and go for it again 👍

**Comment by leewynne** [comment]
Hey @rhymes @Ridhwana it passed!

**Review by Ridhwana — APPROVED** [review]

**Review by fdoxyz — APPROVED** [review]

**Pull request merged**